### ISSUE TITLE
Hoverzoom crashes the AWS Amplify dashboard

### ISSUE THREAD

**Issue by justinAurand** [extesy/hoverzoom#545]
**ATTENTION!** If you don't follow this template then your issue will be immediately closed.

## Description of the problem
Upon logging into the AWS interface and going into the AWS Amplify service, the Amplify dashboard interface is frozen, the page is unresponsive, and eventually the browser window asks if you'd like to wait for the website to respond.

## Url of the page where the problem occurs (not the image url)
https://yourRegionHere.console.aws.amazon.com/amplify/home?region=yourRegionHere#/

## Steps to reproduce the problem (include image url here)
1. Log into your AWS instance
2. Go into the AWS Amplify service
3. Watch the page freeze and eventually crash

## OS version, extension version (Chrome only, Firefox is not supported)
Windows 10 Pro, x64, v10.0.18363

**Comment by G-Rath** [comment]
This happens on the VPC dashboard too. Profiling gives me this:

![image](https://user-images.githubusercontent.com/3151613/92431925-cc46f680-f1ec-11ea-846e-038bf57b206c.png)

It seems that the RegExp `/(.*)\/(.*?)\.([^\/]*?)\.(gif|jpe?g|png)$/` is causing problems.

I've not yet compared this to the source here, so this might have changed somewhat, but currently this is the script that's causing problems:

```
var hoverZoomPlugins = hoverZoomPlugins || [];
hoverZoomPlugins.push({
    name:'Amazon',
    version:'0.3',
    prepareImgLinks:function (callback) {
        var res = [];

        var reFullsize = /(.*)\/(.*?)\.([^\/]*?)\.(gif|jpe?g|png)$/;
        var reReplace = '$1/$2.$4';

        hoverZoom.urlReplace(res,
            'img[src]',
            reFullsize,
            reReplace
        );

        $('[style*=url]').each(function() {
            // extract url from style
            var backgroundImage = this.style.backgroundImage;
            var reUrl = /.*url\s*\(\s*(.*)\s*\).*/i
            backgroundImage = backgroundImage.replace(reUrl, '$1');
            // remove leading & trailing quotes
            var backgroundImageUrl = backgroundImage.replace(/^['"]/, '').replace(/['"]+$/, '');
            var fullsizeUrl = backgroundImageUrl.replace(reFullsize, reReplace);
            if (fullsizeUrl != backgroundImageUrl) {
                var link = $(this);
                if (link.data().hoverZoomSrc == undefined) { link.data().hoverZoomSrc = [] }
                if (link.data().hoverZoomSrc.indexOf(fullsizeUrl) == -1) {
                    link.data().hoverZoomSrc.unshift(fullsizeUrl);
                    res.push(link);
                }
            }
        });

        callback($(res));
    }
});
```

This comes from `plugins/amazon.js` apparently.

I've not looked into this further, so it could be that the regexp is unoptimized, but I suspect it'll be something like these dashboards have a lot of something etc etc.

So far I've only had this on the VPC service page(s) (i.e EC2, S3, RDS, Cloudwatch to name a few all are fine)/

**Comment by extesy** [comment]
I'm not sure what's going on but as a workaround you can disable hoverzoom specifically for AWS dashboards either by clicking on the extension icon or through the options.

**Comment by G-Rath** [comment]
Yeah, thats what I've done for now.

If I get a chance I'll have a look into this - the only element that matches the `$('[style*=url]')` selector seems to be for the loading spinner; I'm wondering if maybe those dashboards pull in the spinner a lot or something?

**Comment by marbet** [comment]
I am experiencing identical behavior but on Amazon.com product pages. On occasions when the page finally becomes responsive and completes loading, the affected object is indicated in the network panel (Chrome):

![image](https://user-images.githubusercontent.com/16416606/93261702-f00fbc80-f757-11ea-8265-1964ce020013.png)

Only workaround is what @extesy mentioned. Excluding Amazon via the Hover Zoom+ Sites option has no effect.

All product pages appear to be affected but this is one example I was working with: https://www.amazon.com/gp/product/B00FF8KQJC/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

**Comment by GrosPoulet** [comment]
Solved by #727 
@G-Rath is right: Amazon's spinner triggers the issue here, but it could have happened on any website with (very) large embedded images (base64)
